cmd/nerdctl/container_logs_test.go

```diff
@@ -206,3 +207,30 @@ func TestLogsWithForegroundContainers(t *testing.T) {
 		}(t)
 	}
 }
+
+func TestTailFollowRotateLogs(t *testing.T) {
+	t.Parallel()
+	if runtime.GOOS == "windows" {
+		t.Skip("tail log is not supported on Windows")
+	}
+	base := testutil.NewBase(t)
+	containerName := testutil.Identifier(t)
+
+	const sampleJSONLog = `{"log":"A\n","stream":"stdout","time":"2024-04-11T12:01:09.800288974Z"}`
+	const linesPerFile = 100
+
+	defer base.Cmd("rm", "-f", containerName).Run()
+	base.Cmd("run", "-d", "--log-driver", "json-file",
+		"--log-opt", fmt.Sprintf("max-size=%d", len(sampleJSONLog)*linesPerFile),
+		"--log-opt", "max-file=10",
+		"--name", containerName, testutil.CommonImage,
+		"sh", "-euc", "while true; do echo A; done").AssertOK()
+
+	tailLogCmd := base.Cmd("logs", "-f", containerName)
+	tailLogCmd.Timeout = 100 * time.Millisecond
+	tailLogs := strings.Split(strings.TrimSpace(tailLogCmd.Run().Combined()), "\n")
```

Comment:
`tailLogCmd.Run()` will block forever because the container keeps writing to stdout.